outlines/generator.py

```diff
@@ -80,22 +74,43 @@ def __init__(self, model, output_type: Optional[Any]):
                 )
 
     @classmethod
-    def from_processor(cls, model, processor: OutlinesLogitsProcessor):
+    def _from_processor(_, subclass, model, processor: OutlinesLogitsProcessor):
         if not isinstance(processor, OutlinesLogitsProcessor):
             raise TypeError("The processor argument must be an instance of OutlinesLogitsProcessor")
-        instance = cls.__new__(cls)
+        instance = subclass.__new__(subclass)
         instance.model = model
         instance.logits_processor = processor
-
         return instance
 
+
+class SteerableGenerator(BaseSteerableGenerator):
+    """Synchronous generator for which we control constrained generation."""
+
+    @classmethod
+    def from_processor(cls, model, processor: OutlinesLogitsProcessor):
+        return super()._from_processor(cls, model, processor)
+
     def __call__(self, prompt, **inference_kwargs):
         return self.model.generate(prompt, self.logits_processor, **inference_kwargs)
 
     def stream(self, prompt, **inference_kwargs):
         return self.model.generate_stream(prompt, self.logits_processor, **inference_kwargs)
 
 
+class AsyncSteerableGenerator(BaseSteerableGenerator):
```

Comment:
Which integration will use  `AsyncSteerableGenerator`?

Reply:
None actually. I changed it back